caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+                            end=sentence_span.end,
+                            word=sentence_span.text,
+                            entity=label,
+                            score=classification.score,
+                        )
+                        token_classification_results.append(token_classification)
+        return TokenClassificationResult(results=token_classification_results)
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        module_saver = ModuleSaver(
+            self,
+            model_path=model_path,
+        )
+
+        with module_saver:
+            module_saver.save_module(self.sentence_splitter, "sentence_split")
+            module_saver.save_module(
+                self.sequence_classifier, "sequence_classification"
+            )
+            config_options = {
+                "language": self.lang,
+                "default_threshold": self.default_threshold,
+                "labels_mapping": self.labels_mapping,
+                "labels_to_output": self.labels_to_output,
+            }
+            module_saver.update_config(config_options)
+
+    @classmethod
+    def load(cls, model_path: str) -> "TransformerSentenceClassification":
+        """Load a transformer sentence classification model.
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            TransformerSentenceClassification
+                Instance of this class built from the on disk model.
+        """
+        config = ModuleConfig.load(os.path.abspath(model_path))
+        loader = ModuleLoader(model_path)
+        sentence_splitter = loader.load_module("sentence_split")
+        try:
+            sequence_classifier = loader.load_module("sequence_classification")
+        except Exception:  # pylint: disable=broad-exception-caught
+            # Module loader looks for config.yml so we load this directly,
+            # pending a way to load HF models directly through config.json
+            # - https://github.com/caikit/caikit/issues/236
+            log.info(
+                "<NLP47789919I>",
+                "Model not able to be loaded directly. \
+                    Currently occurs for models without config.yml",
+            )
+            sequence_classification_path = os.path.join(
+                model_path, config.module_paths["sequence_classification"]
+            )
+            sequence_classifier = SequenceClassification.load(
+                sequence_classification_path
+            )
+        return cls(
+            sentence_splitter=sentence_splitter,
+            sequence_classifier=sequence_classifier,
+            lang=config.language,
+            default_threshold=config.default_threshold,
+            labels_to_output=config.labels_to_output,
+            labels_mapping=config.labels_mapping,
```

Comment:
Since we are not training it, where are we getting the `lan`, `default_threshold`, `labels_to_output` and `labels_mapping` from?

Reply:
Also what is `labels_to_output` ?

Reply:
This is one of the cases where I think for a specific combination of module instances (sentence splitter, classifier), it is definitely possible to just initialize then save the artifact to provide inference. `train` would look like "configure" here as well, and I was hesitant at this time to bring in that debate and separate method at this time.

`labels_to_output` is meant to help in the case where a model is like a "detector" of sorts, so only one label (say label X vs the other label noX) is of interest. Because "detecting" would not apply to all models, this is optional

Reply:
Discussed offline - the module is renamed to `FilteredSpanClassification` to avoid confusion on just `classification`. `labels_mapping` as noted earlier has been removed for now, to be associated with `sequence_classification` as needed in the future